src/languages/kotlin.js

```diff
@@ -95,27 +95,54 @@ export default class Kotlin {
     return "".padStart(level || 0, " ");
   }
 
-  static getRandomFillerLine(indentLevel) {
+  static getRandomFillerLine(indentLevel, memory) {
     const options = [
       () => `${Kotlin.indent(indentLevel)}println(${Helpers.getRandomLogLine()})`,
-      () => Kotlin.getRandomVariableDeclaration(indentLevel),
+      () => Kotlin.getRandomVariableDeclaration(indentLevel, memory),
       () => Kotlin.getRandomFunctionCall(indentLevel, 3),
       () => Kotlin.getRandomLoop(indentLevel || 0),
     ];
     return Helpers.getRandomEntry(options)();
   }
 
   static generateRandomCode(lines) {
-    const firstLine = `${Kotlin.getRandomMethodSignature()} {${Helpers.addNewLine()}`;
+    const memory = new Memory();
+    let methodSignatureReturn = "";
     let fillerLineQty = parseInt(lines, 10) - 2;
     let fillerLines = [];
 
     for (let i = 1; i <= fillerLineQty; i++) {
-      fillerLines.push(`    ${Kotlin.getRandomFillerLine()}`);
+      fillerLines.push(`    ${Kotlin.getRandomFillerLine(0, memory)}`);
+    }
+    let returnLine = "";
+    if (Helpers.getRandomInt(0, 5) > 2) {
+      const randomVariable = memory.getRandomVariable();
+      if (randomVariable !== null) {
+        returnLine = `${Helpers.addNewLine()}    return ${randomVariable.name}`;
+        methodSignatureReturn = `: ${randomVariable.varType.name()}`
+      }
     }
+    const firstLine = `${Kotlin.getRandomMethodSignature()}${methodSignatureReturn} {${Helpers.addNewLine()}`;
+    const lastLine = `${returnLine}${Helpers.addNewLine()}}`;
+    return firstLine + fillerLines.join(Helpers.addNewLine()) + lastLine;
+  }
+}
 
-    const lastLine = `${Helpers.addNewLine()}}`;
+class Memory {
```

Comment:
Would this be good to be adopted by all languages and perhaps moved to the helper?

Reply:
I think it could be very useful but I wasn't sure it was "mature" enough yet.